feature_engine/creation/decision_tree.py

```diff
@@ -0,0 +1,328 @@
+
+        self.variable_combinations_ = self._create_variable_combinations()
+        self.variable_combination_indices_ = {}
+        self.fitted_estimators_ = {}
+
+        for idx, combo in enumerate(self.variable_combinations_):
+            self.variable_combination_indices_[f"estimator_{idx}"] = combo
+            estimator = self._make_decision_tree()
+            self.fitted_estimators_[f"estimator_{idx}"] = estimator.fit(X[combo], y)
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Creates new features using scikit-learn's decision tree.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features]
+            The data to be transformed.
+
+        Returns
+        -------
+        X_new: Pandas dataframe.
+            The original dataframe plus the additional features.
+        """
+        X = super().transform(X)
+        self.new_features_names_ = self._create_new_features_names()
+
+        for idx, estimator in self.fitted_estimators_.items():
+            new_feature = self.new_features_names_[idx]
+            X[new_feature] = estimator.predict(
+                X[self.variable_combination_indices_[idx]]
+            )
+
+        return X
+
+    def _make_decision_tree(self):
+        """Instantiate decision tree."""
+        if self.regression is True:
+            est = DecisionTreeRegressor(max_depth=self.max_depth)
+        else:
+            est = DecisionTreeClassifier(max_depth=self.max_depth)
+
+        return est
+
+    def _create_variable_combinations(self):
+        """
+        Create a list of the different combinations of variables that will be
+        used to create new features.
+        """
+        variable_combinations = []
+        if isinstance(self.output_features, tuple):
+            for feature in self.output_features:
+                if isinstance(feature, str):
+                    variable_combinations.append([feature])
+                else:
+                    variable_combinations.append(list(feature))
+
+        # if output_features is None, int or list.
+        else:
+            if self.output_features is None:
+                for num in range(1, len(self.variables) + 1):
+                    variable_combinations += list(combinations(self.variables, num))
+
+            elif isinstance(self.output_features, int):
+                for num in range(1, self.output_features + 1):
+                    variable_combinations += list(combinations(self.variables, num))
+
+            # output_feature is a list
+            else:
+                for num in self.output_features:
+                    variable_combinations += list(combinations(self.variables, num))
+
+            # transform all elements to lists to slice X dataframe
+            variable_combinations = [list(var) for var in variable_combinations]
+
+        return variable_combinations
+
+    def _create_new_features_names(self):
```

Comment:
for convention rename to get_feature_names_out() and it should also be able to take input_features=None.

It also returns a list instead of a dictionary, not sure how / if that could be made to fit with the current logic?